src/Terrain/Maps/MapReader.java

```diff
@@ -0,0 +1,177 @@
+package Terrain.Maps;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+import Engine.XYCoord;
+import Terrain.Environment;
+import Terrain.MapInfo;
+
+public class MapReader extends IMapBuilder
+{
+  /**
+   * Tells the MapReader to read in the maps.
+   */
+  public static ArrayList<MapInfo> readMapData()
+  {
+    ArrayList<MapInfo> importMaps = new ArrayList<MapInfo>();
+
+    // This try{} is to safeguard us from exceptions if the res/map folder doesn't exist.
+    // If it fails, we don't need to do anything in the catch{} since we just won't have anything in our list.
+    try
+    {
+      final File folder = new File("res/map");
+
+      for( final File fileEntry : folder.listFiles() )
+      {
+        // we aren't checking subdirectories, yet
+        if( !fileEntry.isDirectory() )
+        {
+          try
+          {
+            // We get the filename, and make it look nice for our map list.
+            String mapName = fileEntry.getName();
+            // Mostly, we just don't want to try to interpret the python script as a map. That'd be bad.
+            if( !mapName.contains(".map") )
+              continue;
+            // underscores->spaces makes it pretty
+            mapName = mapName.replaceAll("_", " ");
+            mapName = mapName.replaceAll(".map", "");
+            System.out.println("INFO: Parsing map: " + mapName);
+
+            // We need a list of who starts owning what properties. This is that list.
+            // Each arraylist contains coordinates, and which list it is denotes who owns that property.
+            ArrayList<ArrayList<XYCoord>> properties = new ArrayList<ArrayList<XYCoord>>();
+            // Will support 14 sides in the future, possibly more.
+            for( int i = 0; i < 4; i++ ) // TODO: add more sides
+            {
+              properties.add(new ArrayList<XYCoord>());
+            }
+
+            Scanner scanner = new Scanner(fileEntry);
+            // We need the first line so we can pre-fill our terrain data array with the proper number of sub-arrays.
+            String line = scanner.nextLine();
+
+            // The game's internal representation of the map is transposed from the representation you'd see if you wrote out the array.
+            // This is so because that allows you to access mapArray[x][y], which is convenient for handling the map.
```

Comment:
I'm not sure this is any more clear. I think the default expectation going in is that the third tile from the left and two down (so, location 2, 1), would correspond to mapArray[2][1]. Adding a bunch of verbiage here about transposed matrices is just going to create confusion, and subvert that initial (and valid) expectation. The explanation here isn't super clear either, which doesn't help. If anything, I'd just point out that each subarray is a column, and rows span across the subarrays at a given index, and if we want to talk about how manually writing out a 2D array in code might look different than it does when drawn out, we can do that in some location where we are actually manually writing out a 2D array in code.

Reply:
I suppose I was more explaining it to myself than anyone else, and that's not terribly relevant since the code works.
I cut out most of it.